### ISSUE TITLE
No error or warnings with not inited variables

### ISSUE THREAD

**Issue by ziggi** [pawn-lang/compiler#420]
### Issue description:

Look at the code, I think compiler should give an error or warning.

### Minimal complete verifiable example (MCVE):

```pawn
main()
{
	new test = test;
	printf("test = %d", test);
}
```

Results: `test = 1`

```pawn
main()
{
	new arr[3] = {23, 123, 223};
	new test = arr[test];
	printf("test = %d", test);
}
```

Results: `test = 123`

### Workspace Information:

* Compiler version:  3.10.9
* Command line arguments provided (or sampctl version): ` -;+ -(+ -d2`
* Operating System: `Windows`

**Comment by YashasSamaga** [comment]
It should be an `undefined symbol` error. Logically, the expression is evaluated first where `test` is not yet defined.
